testsuite/openshift/objects/auth_config.py

```diff
@@ -19,6 +19,22 @@ class MatchExpression:
     key: str = "group"
 
 
+@dataclass
+class Rule:
```

Comment:
Just nitpick, since we also define an interface, it should be present there and not only in our implementation(=move it to objects/__init__.py :D)